lib/dal/src/cached_module.rs

```diff
@@ -430,10 +433,11 @@ impl CachedModule {
         Ok(result)
     }
 
-    pub async fn get_user_scoped_modules(
```

Comment:
Before, get_user_scoped_modules() got you *only* user-scoped modules, and you appended that to non-scoped modules. Now, it's just `latest_modules()` and gets both (by retrieving the user from the context).

latest_modules() used to get only non-scoped modules; now all callers of that use latest_user_independent_modules(). It feels like most callers should be using the new latest_modules() and including user-scoped modules, but I don't know enough to be sure they're not excluding them on purpose.